backend/app/main/views.py

```diff
@@ -1,10 +1,111 @@
 import datetime
-from flask import Flask, jsonify, request, abort, make_response
+from flask import Flask, jsonify, request, abort, make_response, url_for, session
+from twilio.twiml.messaging_response import MessagingResponse
+
 from . import main
 from .. import db
+from .models import Survey, Question, Answer 
 
 
 # get all users
 @main.route("/", methods=["GET"])
 def index():
     return "Hello World!"
+
+
+# main control flow; direct user to welcome message or next question
+@main.route("/message")
+def sms_signup():
+    response = MessagingResponse()
+
+    signup_survey = Survey.query.first()
+    if survey_error(signup_survey, response.message):
+        return str(response)
+
+    session['id'] = request.values['From']
+
+    if 'question_id' in session:
+        response.redirect(url_for('main.answer',
+	                          question_id=session['question_id']))
+    elif request.values.get('Body', None) == 'MOVEUP':
+        redirect_to_first_question(response, signup_survey)
+    else:
+        welcome_user(signup_survey, response.message)
+    return str(response)
+
+
+@main.route('/question/<question_id>')
```

Comment:
It's probably good to add the type(s) of HTTP methods this route accepts (i.e. GET), similar to line 11 or line 44